src/server/util/WatcherList.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ */
+
+const watcherList = [];
+
+const addWatcher = watcher => {
+  watcherList.push(watcher);
+  return watcherList.length - 1;
+};
+
+const removeWatcher = index => {
+  watcherList.splice(index, 1);
+};
+
+const getList = () => watcherList;
+
+const clearList = () => {
+  watcherList.length = 0;
+};
+
+module.exports = {
```

Comment:
You get my points by writing this abstraction, but in case of this PR, I believe previous version using an array and primitive operation was a bit better.